.circleci/config.yml

```diff
@@ -0,0 +1,173 @@
+            |  tar xj
+      - run:
+          name: Install geckodriver
+          command: >
+            curl --retry 5 -LO https://github.com/mozilla/geckodriver/releases/download/v0.21.0/geckodriver-v0.21.0-linux64.tar.gz
+            && tar xf geckodriver-v0.21.0-linux64.tar.gz
+
+      # Run tests
+      - run:
+          name: Run all tests
+          command: GECKODRIVER=$(pwd)/geckodriver PATH=$(pwd)/firefox:$PATH ./test.sh
+
+      # Save cache
+      - save_cache:
+          key: v1-cargo-cache-test-{{ arch }}-{{ .Branch }}
+          paths:
+            - target
+            - /usr/local/cargo
+      - save_cache:
+          key: v1-cargo-cache-test-{{ arch }}
+          paths:
+            - target
+            - /usr/local/cargo
+
+  docs-build:
+    docker:
+      - image: rust:latest
+    steps:
+      - checkout
+      # Load cargo target from cache if possible.
+      # Multiple caches are used to increase the chance of a cache hit.
+      - restore_cache:
+          keys:
+            - v1-cargo-cache-docs-{{ arch }}-{{ .Branch }}
+            - v1-cargo-cache-docs-{{ arch }}
+
+      # Install nightly
+      - run:
+          name: Install Rust nightly
+          command: rustup update nightly && rustup default nightly
+
+      # Show versions
+      - run:
+          name: Show versions
+          command: rustc --version && cargo --version
+
+      # Install mdbook
+      - run:
+          name: Install mdbook
+          command: >
+            (test -x $CARGO_HOME/bin/cargo-install-update || cargo install cargo-update)
+            && (test -x $CARGO_HOME/bin/mdbook || cargo install --vers "^0.1" mdbook)
+            && mv ~/.gitconfig ~/.gitconfig.disabled  # Workaround for https://github.com/nabijaczleweli/cargo-update/issues/100
+            && cargo install-update -a
+            && mv ~/.gitconfig.disabled ~/.gitconfig
+
+      # Build docs
+      - run:
+          name: Build docs
+          command: >
+            (cd book && mdbook build)
+            && cargo doc --no-deps -p virtual-dom-rs -p css-rs-macro -p html-macro
+            && cp -R target/doc book/book/api
+      - persist_to_workspace:
+          root: book
+          paths: book 
+
+      # Save cache
+      - save_cache:
+          key: v1-cargo-cache-docs-{{ arch }}-{{ .Branch }}
+          paths:
+            - target
+            - /usr/local/cargo
+      - save_cache:
+          key: v1-cargo-cache-docs-{{ arch }}
+          paths:
+            - target
+            - /usr/local/cargo
+
+  docs-deploy:
+    docker:
+      - image: node:10
+    steps:
+      - checkout
+      - attach_workspace:
+          at: book
+      - run:
+          name: Disable jekyll builds
+          command: touch book/book/.nojekyll
+      - run:
+          name: Install and configure dependencies
+          command: >
+            npm install -g --silent gh-pages@2
```

Comment:
Why silent?

Reply:
Less noise. But we can also remove that part, it's captured anyways :slightly_smiling_face: